### PR TITLE
chore(docs): update code of conduct of synthtool and templates

### PR DESCRIPTION
**Pull request by crwilcox**
Replace v1.2.0 with v1.4.0 of https://www.contributor-covenant.org/version/1/4/code-of-conduct/

### CODE REVIEW

**Comment by crwilcox** [comment]
Figured out why I didn't see project stewards. The Open Source Program has modified 1.4 slightly. Change coming.

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -1,43 +1,93 @@
...
+to engage in behavior that violates the project’s code of conduct.
+
+If you see someone violating the code of conduct, you are encouraged to address
+the behavior directly with those involved. Many issues can be resolved quickly
+and easily, and this gives people more control over the outcome of their
+dispute. If you are unable to resolve the matter for any reason, or if the
+behavior is threatening or harassing, report it. We are dedicated to providing
+an environment where participants feel welcome and safe.
+
+Reports should be directed to *[PROJECT STEWARD NAME(s) AND EMAIL(s)]*, the
```
Comment (coryan) [inline CODE_OF_CONDUCT.md:72]: I think you are supposed to name a steward for this project, and put their name and email here.

**Review by coryan — COMMENTED** [review]
I take it these templates are pushed automatically when a new project is created?  Consider (for a future PR): maybe the tool can ask for a steward, update the template when the project is created, and push the code of conduct file with a name already filled out.

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (crwilcox) [reply CODE_OF_CONDUCT.md:72]: Thanks @coryan . We are chatting as a team how to manage this (if we want it to be the same across all libraries in a project or if we should support templating)

**Comment by crwilcox** [comment]
This is ready for review, files have had string placeholders replaced.

**Review by busunkim96 — APPROVED** [review]

**Review by bcoe — APPROVED** [review]

**Review by SurferJeffAtGoogle — APPROVED** [review]

**Comment by SurferJeffAtGoogle** [comment]
@chingor13,  this PR needs an approval from googleapis/yoshi-java.

**Review by chingor13 — APPROVED** [review]

**Pull request merged**